npm/private/npm_translate_lock.bzl

```diff
@@ -131,6 +132,34 @@ def _gather_values_from_matching_names(keyed_lists, *names):
                 result.append(v)
     return result
 
+def _get_npm_auth(rctx):
+    _NPM_TOKEN_KEY = "//registry.npmjs.org/:_authtoken"
+    token = None
+
+    # Read token from npmrc label
+    if rctx.attr.npmrc:
+        npmrc_path = rctx.path(rctx.attr.npmrc)
+        npmrc = parse_ini(rctx.read(npmrc_path))
+        if _NPM_TOKEN_KEY in npmrc:
```

Comment:
Unless I'm reading something very wrong, this only allows authentication to the npmjs registry itself. What about arbitrary private repositories like AWS CodeArtifact or similar?

Reply:
Yap, unfortunately this only allows authentication to the npmjs registry. 
Do you have any suggestion on how we could support 3rd party registries and private repositories? It has to be agnostic to the registry so I'm worried that we would need to calculate the appropriate token for each package rather than having a single token for all packages.

Reply:
I've created #499 to address this issue. 
Let me know what you think about it.